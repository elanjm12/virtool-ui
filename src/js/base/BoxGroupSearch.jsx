import React from "react";
import styled from "styled-components";
import { BoxGroupSection } from "./Box";
import { Icon } from "./Icon";
import { Input } from "./Input";

const IconContainer = styled.div`
align-items: center;
display: flex;
    margin-top: 10px;
    position: absolute;
    right 20px;
`;

const StyledBoxGroupSearch = styled(BoxGroupSection)`
    display: flex;
    padding: 10px;
    position: relative;
`;

export const BoxGroupSearch = ({ label, placeholder, value, onChange, autoFocus = false }) => (
    <StyledBoxGroupSearch>
        <Input
            value={value}
            placeholder={placeholder}
            aria-label={label}
            onChange={e => onChange(e.target.value)}
            autoFocus={autoFocus}
        />
        <IconContainer>
            <Icon name="times fa-fw" onClick={() => onChange("")} aria-label="clear" />
        </IconContainer>
    </StyledBoxGroupSearch>
);
