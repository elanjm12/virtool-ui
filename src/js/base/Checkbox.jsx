import * as RadixCheckbox from "@radix-ui/react-checkbox";
import React from "react";
import { Icon } from "./Icon";
import styled from "styled-components";

const getBackgroundColor = ({ checked, theme }) => theme.color[checked ? "primary" : "transparent"];
const getBorder = ({ checked, theme }) => (checked ? "none" : `2px solid ${theme.color.greyDark}`);
const getColor = ({ checked, theme }) => theme.color[checked ? "white" : "grey"];

const CheckboxContainer = styled.div`
    display: inline-flex;
`;

export const CheckboxLabel = styled.span`
    vertical-align: bottom;
    cursor: pointer;
    margin-left: 5px;
    user-select: none;
`;

export const StyledCheckbox = styled(RadixCheckbox.Root)`
    align-items: center;
    background-color: ${getBackgroundColor};
    border: ${getBorder};
    border-radius: ${props => props.theme.borderRadius.sm};
    color: ${getColor};
    cursor: pointer;
    display: inline-flex;
    font-size: 11px;
    height: 20px;
    justify-content: center;
    margin-right: 0;
    opacity: ${props => (props.checked ? 1 : 0.5)};
    transition: opacity 100ms ease;
    width: 20px;
`;

export const Checkbox = ({ checked, label, onClick, disabled }) => (
    <CheckboxContainer>
        <StyledCheckbox
            disabled={disabled}
            checked={checked}
            onClick={disabled ? null : onClick}
            aria-label={label || "checkbox"}
        >
            <Icon checked={checked} name="check" />
            <RadixCheckbox.Indicator />
        </StyledCheckbox>
        {label ? <CheckboxLabel onClick={disabled ? null : onClick}>{label}</CheckboxLabel> : null}
    </CheckboxContainer>
);

Checkbox.defaultProps = {
    checked: false,
    disabled: false
};
