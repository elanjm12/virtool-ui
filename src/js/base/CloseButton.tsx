import React from "react";
import styled from "styled-components";
import { Icon } from "./Icon";

const StyledButton = styled(Icon)`
    color: inherit;
`;

export const CloseButton = ({ onClick }) => <StyledButton aria-label="close" name="times" onClick={onClick} />;
