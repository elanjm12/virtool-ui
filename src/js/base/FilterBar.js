import React from "react";
import styled from "styled-components";
import { Input } from "./Input";
import { Icon } from "./Icon";

const SearchbarContainer = styled.div`
    display: flex;
    position: relative;
    width: 100%;
    padding: 10px;
`;

const SearchbarClearIcon = styled(Icon)`
    position: absolute;
    padding-top: 12px;
    right: 20px;
`;

export const FilterBar = ({ term, setTerm, type, autoFocus }) => (
    <SearchbarContainer>
        <Input
            value={term}
            placeholder={`Filter ${type}`}
            aria-label={`Filter ${type}`}
            onChange={e => setTerm(e.target.value)}
            autoFocus={autoFocus}
        />

        <SearchbarClearIcon name="times fa-fw" onClick={() => setTerm("")} aria-label="clear" />
    </SearchbarContainer>
);
