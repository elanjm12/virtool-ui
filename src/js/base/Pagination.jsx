import React from "react";
import styled from "styled-components";
import { map, max, min, range } from "lodash-es";
import { Link } from "react-router-dom";
import { LinkButton } from "./Button";

import { getColor, getFontSize } from "../app/theme";

const StyledPaginationBox = styled.div`
    display: flex;
    width: 500px;
    justify-content: center;
    margin-left: 25%;
    align-items: center;
`;

const StyledLink = styled(Link)`
    margin: 5px;
    font-size: ${getFontSize("lg")};
    pointer-events: ${props => (props.disabled ? "none" : "")};
    color: ${props => getColor({ color: props.$active ? "blue" : "blueDarkest", theme: props.theme })};
`;

const StyledNextLink = styled(LinkButton)`
    display: flex;
    margin: 5px;
    width: 60px;
    justify-content: center;
    pointer-events: ${props => (props.disabled ? "none" : "")};
`;

const StyledPaginationItemsContainer = styled.div`
    padding-bottom: 5px;
`;

const getPageRange = (pageCount, storedPage, leftButtons = 1, rightButtons = 2) => {
    const totalButtons = leftButtons + rightButtons;
    let maxVal = min([pageCount, storedPage + rightButtons]);
    const minVal = max([1, maxVal - totalButtons]);
    maxVal = min([pageCount, minVal + totalButtons]);

    return range(minVal, maxVal + 1);
};

export const Pagination = ({ items, renderRow, storedPage, currentPage, pageCount, onLoadNextPage }) => {
    const entries = map(items, item => renderRow(item));

    const pageButtons = map(getPageRange(pageCount, storedPage), pageNumber => (
        <StyledLink
            key={pageNumber}
            to={`?page=${pageNumber}`}
            $active={storedPage !== pageNumber}
            disabled={storedPage === pageNumber}
            onClick={() => onLoadNextPage(pageNumber)}
        >
            {pageNumber}
        </StyledLink>
    ));

    return (
        <div>
            <StyledPaginationItemsContainer>{entries}</StyledPaginationItemsContainer>
            {pageCount > 1 && (
                <StyledPaginationBox>
                    <StyledLink
                        to={`?page=${currentPage - 1}`}
                        color="blue"
                        disabled={currentPage === 1}
                        $active={currentPage !== 1}
                        onClick={() => onLoadNextPage(currentPage - 1)}
                    >
                        Previous
                    </StyledLink>
                    {pageButtons}
                    <StyledNextLink
                        to={`?page=${currentPage + 1}`}
                        color="blue"
                        disabled={currentPage === pageCount}
                        onClick={() => onLoadNextPage(currentPage + 1)}
                    >
                        Next
                    </StyledNextLink>
                </StyledPaginationBox>
            )}
        </div>
    );
};
