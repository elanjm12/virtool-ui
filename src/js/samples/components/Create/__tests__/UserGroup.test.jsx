import { Select } from "../../../../base";
import { SampleUserGroup } from "../UserGroup";
import { vi } from "vitest";

describe("SampleUserGroup", () => {
    let props;
    beforeEach(() => {
        props = {
            groups: ["foo"],
            onChange: vi.fn(),
            group: "bar"
        };
    });

    it("should render", () => {
        const wrapper = shallow(<SampleUserGroup {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should call onChange when InputError is changed", () => {
        const e = {
            target: "foo"
        };
        const wrapper = shallow(<SampleUserGroup {...props} />);
        wrapper.find(Select).simulate("change", e);
        expect(props.onChange).toHaveBeenCalledWith({ target: "foo" });
    });
});
