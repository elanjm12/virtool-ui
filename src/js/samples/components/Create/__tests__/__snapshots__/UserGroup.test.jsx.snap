// Vitest Snapshot v1

exports[`SampleUserGroup > should render 1`] = `
<InputGroup>
  <styled.label>
    User Group
  </styled.label>
  <Select
    onChange={[MockFunction spy]}
    value="bar"
  >
    <option
      key="none"
      value="none"
    >
      None
    </option>
    <styled.option
      key="foo"
      value="foo"
    >
      foo
    </styled.option>
  </Select>
</InputGroup>
`;
